### ISSUE TITLE
Product Price: Issues like no space between normal price and sale price

### ISSUE THREAD

**Issue by imanish003** [woothemes/woocommerce#49102]
I found a few minor issues in "Product Price" block while testing on WordPress 6.6. Here are the details:

**Issue 1:** There is no space between the normal price and the sale price. Here is how to reproduce it:

1. In WooCommerce > Settings > General, this is what I had set that resulted in the issue:
  - Currency: AFN - Afghan afghani
  - Currency position: Left with space
  - Thousand separator: .
  - Decimal separator: ,
  - Number of decimals: 2
2. Create a new post and add Product Collection block. In "Product Price" block, you will see that there is no space between normal price and sale price as shown in the screenshot below.
![image](https://github.com/woocommerce/woocommerce/assets/16707866/f0ca6ef8-9d15-42d1-b271-55922a483e37)

**Issue 2:** It's looking weird when I keep the same settings as above but change "Currency position" to "Right with space". Here is how to reproduce it:

1. In WooCommerce > Settings > General, change "Currency position" to "Right with space".
2. Create a new post and add Product Collection block. In "Product Price" block, you will notice that it's looking weird as shown in the screenshot below.
![image](https://github.com/woocommerce/woocommerce/assets/16707866/1fe2fbd5-1ab9-42db-bdd4-a0239d3ecf4c)

**Comment by AhmarZaidi** [comment]
Hey @imanish003 

I've replicated the issue. 

I believe it's comes from how the `NumberFormat` component from 'react-number-format' handles RTL characters in its prefix.

Even though we provide `؋` in the prefix it is rendered after the `priceValue`. 

One way we can fix this is by using the [`unicode-bidi`](https://developer.mozilla.org/en-US/docs/Web/CSS/unicode-bidi) along with [`direction`](https://developer.mozilla.org/en-US/docs/Web/CSS/direction) CSS properties.

Here's how it looks in the editor after the changes:

https://github.com/user-attachments/assets/e5de2d19-493d-4de8-bdd4-96553ed5adc6


I'd like to work on this issue, let me know if this approach looks correct.

Thanks!

**Comment by AhmarZaidi** [comment]
Hey @imanish003,

I've noticed that this issue appears everywhere we are using `FormattedMonetaryAmount` (which is a wrapper around `NumberFormat`)

Here are a few examples with currency position: `Left with space`

  - Checkout page order summary
    <img width="500" alt="image" src="https://github.com/user-attachments/assets/ee31dc63-6190-4d80-b986-0a6a94ce3c30" />

  - Cart page
    <img width="500" alt="image" src="https://github.com/user-attachments/assets/e68b2657-02f7-4251-8a58-d816ca80a272" />

  - Variable product frontend
    <img width="500" alt="image" src="https://github.com/user-attachments/assets/cd5cfa72-ba7e-42e6-93ef-9fcf32820cd4" />

**Comment by AhmarZaidi** [comment]
Hey @imanish003 @nerrad,

Could you please review the approach/observations outlined above for this issue when available?
I'd like to work on it and want to confirm if the direction looks good before opening a PR.

Thanks!

**Comment by AhmarZaidi** [comment]
Hey @nerrad ,

Wondering if this issue is still open for contribution for me.
I have a PR ready for the initial issue and I was also working on the fix for additional issue I reported https://github.com/woocommerce/woocommerce/issues/49102#issuecomment-2865305538

Thanks!

**Comment by tjcafferkey** [comment]
@AhmarZaidi yes of course, we would welcome a PR on this issue if you'd like to submit one that would be great.

**Comment by AhmarZaidi** [comment]
Hey @tjcafferkey,

I've raised a draft PR: https://github.com/woocommerce/woocommerce/pull/58134

**Comment by kkoteen** [comment]
Hi @imanish003 - thanks for reporting this, and for the detail you gave.

I know this has been open a while, and I appreciate your patience. We're currently going through some older WooCommerce issues, identifying what still needs fixing, and tidying up what's already fixed or out of scope.

It looks like this has already been fixed, or it's no longer happening on trunk.

Here's what I found: Current monetary formatting code on trunk includes the RTL currency-symbol fix that addressed the spacing and ordering problems reported for Product Price and other surfaces using FormattedMonetaryAmount.

I'm going to close this one. If you're still facing this issue on a current version, please reply with steps to reproduce and we'll happily take another look.

Fix/changelog reference: https://github.com/woocommerce/woocommerce/pull/58134; https://github.com/woocommerce/woocommerce/pull/61000; c4c8b667eb Reapply Fix rendering issue with RTL currency symbols in product price block (#58134) (#61000)
